diff --git a/packages/dev-env/src/bsky.ts b/packages/dev-env/src/bsky.ts
index c0362bc45..765035e0a 100644
--- a/packages/dev-env/src/bsky.ts
+++ b/packages/dev-env/src/bsky.ts
@@ -94,6 +94,8 @@ export class TestBsky {
       etcdHosts: [],
     })
 
+    console.log("################# TestBsky::create(cfg, config) ", cfg, config)
+
     // Separate migration db in case migration changes some connection state that we need in the tests, e.g. "alter database ... set ..."
     const migrationDb = new bsky.Database({
       url: cfg.dbPostgresUrl,
diff --git a/packages/dev-env/src/bsync.ts b/packages/dev-env/src/bsync.ts
index 1d69c6a39..7305f71c7 100644
--- a/packages/dev-env/src/bsync.ts
+++ b/packages/dev-env/src/bsync.ts
@@ -23,6 +23,7 @@ export class TestBsync {
       ...cfg,
     })
 
+    console.log("############ TestBsync::create(cfg, config) ", cfg, config)
     const service = await bsync.BsyncService.create(config)
     await service.ctx.db.migrateToLatestOrThrow()
     await service.start()
diff --git a/packages/dev-env/src/introspect.ts b/packages/dev-env/src/introspect.ts
index fc9657900..fc431cb95 100644
--- a/packages/dev-env/src/introspect.ts
+++ b/packages/dev-env/src/introspect.ts
@@ -28,6 +28,35 @@ export class IntrospectServer {
   ) {
     const app = express()
     app.get('/', (_req, res) => {
+      res.status(200).send({
+        plc: {
+          url: plc.url,
+        },
+        pds: {
+          url: pds.url,
+          did: pds.ctx.cfg.service.did,
+	  ctxcfg: pds.ctx.cfg
+        },
+        bsync: {
+          url: bsync.url,
+	  ctxcfg: bsync.ctx.cfg
+        },
+        bsky: {
+          url: bsky.url,
+          did: bsky.ctx.cfg.serverDid,
+	  ctxcfg: bsky.ctx.cfg
+        },
+        ozone: {
+          url: ozone.url,
+          did: ozone.ctx.cfg.service.did,
+	  ctxcfg: ozone.ctx.cfg
+        },
+        db: {
+          url: ozone.ctx.cfg.db.postgresUrl,
+        },
+      })
+    })
+    app.get('/origin', (_req, res) => {
       res.status(200).send({
         plc: {
           url: plc.url,
